query-builder/src/main/java/com/datastax/oss/driver/internal/querybuilder/update/DefaultUpdate.java

```diff
@@ -154,6 +180,19 @@ public String asCql() {
       }
     }
 
+    if (ttlInSeconds != null) {
+      // choose the correct keyword based on whether or not we have a timestamp
+      builder.append((timestamp != null) ? " AND " : " USING ").append("TTL ");
+      if (ttlInSeconds instanceof BindMarker) {
+        ((BindMarker) ttlInSeconds).appendTo(builder);
+      } else if (ttlInSeconds instanceof Integer) {
+        builder.append(ttlInSeconds);
+      } else {
+        throw new IllegalArgumentException(
+            "TTL value must be an Integer greater than or equal to 0");
```

Comment:
I would move the check to the constructor, there's no point in letting the user build an object that will fail later. While we're at it we can do the same for the timestamp.
Nit: the message should say Integer or BindMarker.